File: src/create_ticket.py
import os
import json
import requests
from dotenv import load_dotenv
from requests.auth import HTTPBasicAuth

load_dotenv()
email = os.getenv('ATLASSIAN_EMAIL')
api_token = os.getenv('ATLASSIAN_API_TOKEN')

url = "https://anushachitranshi.atlassian.net/rest/api/2/issue"


headers = {
  "Accept": "application/json",
  "Content-Type": "application/json"
}

payload = json.dumps( {
  "fields": {
        "project": {
           "key": "SCRUM"
        },
        "issueType": {
            "name": "Task"
        },
        "assignee": {
        "id": "5b109f2e9729b51b54dc274d"
        },
        "project": {
        "id": "10000"
        },
        "reporter": {
        "id": "5b10a2844c20165700ede21g"
        },
    },
    "update": {
        "worklog": [
        {
            "add": {
            "started": "2019-07-05T11:05:00.000+0000",
            "timeSpent": "60m"
            }
        }
        ]
  }
} )

auth = HTTPBasicAuth(email, api_token)

response = requests.post(
   url,
   data=payload,
   headers=headers,
   auth=auth,
   verify=r"C:\Users\ac39920\AppData\Local\Programs\Python\Python313\Lib\site-packages\certifi\cacert.pem"
)

print(response.text)

File: src/create_ticket.ts
import * as vscode from 'vscode';
import axios from 'axios';
import * as dotenv from 'dotenv';

// Load environment variables from .env file
dotenv.config();
const email = process.env.ATLASSIAN_EMAIL;
const apiToken = process.env.ATLASSIAN_API_TOKEN;

if (!email || !apiToken) {
    throw new Error("Missing JIRA credentials: Please set JIRA_EMAIL and JIRA_API_TOKEN in the .env file.");
}

export default async function createJiraTicket() {
    // Sample data, you may replace this with actual user input
    const title = await vscode.window.showInputBox({ prompt: 'Enter ticket title' });
    const description = await vscode.window.showInputBox({ prompt: 'Enter ticket description' });
    const assignee = await vscode.window.showInputBox({ prompt: 'Enter assignee' });
    const reporter = await vscode.window.showInputBox({ prompt: 'Enter reporter' });

    const jiraApiUrl = "https://anushachitranshi.atlassian.net/rest/api/2/issue";

    // Sample JIRA ticket payload
    const payload = {
        fields: {
            project: {
                key: 'SCRUM', 
            },
            summary: title || 'Default Title',
            description: description || 'Default Description',
            issuetype: {
                name: 'Task', // Change as needed
            },
            assignee: {
                name: assignee || 'defaultAssignee', // Assignee field
            },
            reporter: {
                name: reporter || 'defaultReporter', // Reporter field
            },
        },
    };
    vscode.window.showInformationMessage('Running test coverage analysis...');
    // API Call
    try {
        const response = await axios.post(jiraApiUrl, payload, {
            auth: {
                username: email || "", // Load from environment variable
                password: apiToken || "", // Load from environment variable
            },
            headers: {
                'Content-Type': 'application/json',
            },
        });
        vscode.window.showInformationMessage(`JIRA ticket created successfully: ${response.data.key}`);
        
    } catch (error: unknown) { // Explicitly typing error as unknown
        if (axios.isAxiosError(error)) {
            // Handle axios error
            vscode.window.showErrorMessage(`Error creating JIRA ticket: ${error.response?.data?.message || error.message}`);
        } else {
            // Handle generic error
            vscode.window.showErrorMessage(`Error creating JIRA ticket: ${String(error)}`);
        }
    }
    }
